File: app/src/main/java/cn/intret/app/picgo/model/SortOrder.java
package cn.intret.app.picgo.model;

public enum SortOrder {
    DESC,
    ASC,
    UNKNOWN;
    public static SortOrder fromString(String string) {
        if (string == null) {
            return UNKNOWN;
        }
        switch (string) {
            case "DESC": return DESC;
            case "ASC": return ASC;
        }
        return UNKNOWN;
    }

    @Override
    public String toString() {
        switch (this) {
            case ASC: return "ASC";
            case DESC: return "DESC";
        }
        return "UNKNOWN";
    }
}

File: app/src/main/java/cn/intret/app/picgo/model/user/data/SortOrder.kt
package cn.intret.app.picgo.model.user.data

/**
 * Image list sort order
 */
enum class SortOrder {
    DESC,
    ASC,
    UNKNOWN;

    override fun toString(): String {
        when (this) {
            ASC -> return "ASC"
            DESC -> return "DESC"
        }
        return "UNKNOWN"
    }

    companion object {
        fun fromString(string: String?): SortOrder {
            if (string == null) {
                return UNKNOWN
            }
            when (string) {
                "DESC" -> return DESC
                "ASC" -> return ASC
            }
            return UNKNOWN
        }
    }
}
